### PR TITLE
Fixes mechs using occupant spacemove

### LINKED ISSUE #11339: Magboots work in mechs
no more zero-g drifting inside mechs!

### PR DESCRIPTION
**Pull request by AnturK**
Basically a copy of mob process spacemove. Mob mechs would be nice.

Fixes  #11339

### CODE REVIEW

**Comment by RemieRichards** [comment]
I smell copypaste.

**Comment by AnturK** [comment]
It is, i guess i could put the object search into a proc but that really wouldn't fix the issue

**Comment by RemieRichards** [comment]
@AnturK 
```
/atom/proc/find_spacemove_backup()
    do_the_thing()
```

overriden for atoms/mobs etc. that have other things that allow them to spacemove

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -406,9 +406,31 @@
...
+	for(var/atom/A in orange(1, get_turf(src)))
+		if(isarea(A))
+			continue
+		else if(isturf(A))
+			var/turf/turf = A
+			if(!turf.density)
+				continue
+			return 1
+		else
+			var/atom/movable/AM = A
```
Comment (RemieRichards) [inline code/game/mecha/mecha.dm]: Wowee this isn't safe.

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -406,9 +406,31 @@
...
+			if(!AM.CanPass(src) || AM.density)
+				if(AM.anchored)
+					return 1
+				dense_object_backup = AM
```
Comment (RemieRichards) [inline code/game/mecha/mecha.dm]: should break after this.

File: code/game/mecha/mecha.dm [hunk 3]
```diff
@@ -406,9 +406,31 @@
...
 /obj/mecha/Process_Spacemove(var/movement_dir = 0)
-	if(occupant)
-		return occupant.Process_Spacemove(movement_dir) //We'll just say you used the clamp to grab the wall
-	return ..()
+	. = ..()
+	if(.)
+		return 1
+
+	var/atom/movable/dense_object_backup
+	for(var/atom/A in orange(1, get_turf(src)))
```
Comment (RemieRichards) [inline code/game/mecha/mecha.dm]: istypeless for loop please.

**Comment by AnturK** [comment]
That's what i meant. Still pretty shit imo.

**Comment by RemieRichards** [comment]
@AnturK better than copy pasta.

**Pull request merged**